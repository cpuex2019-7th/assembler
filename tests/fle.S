    li t1, 0x3f666666           ; 0.9
    fmvwx ft1, t1
    li t2, 0x40400000           ; 3.0
    fmvwx ft2, t2        
    fle t0, ft1, ft2            ; 0.9 <= 3.0 ... t0 should be 1

    li t2, 0x3f666666           ; 0.9
    fmvwx ft2, t2        
    fle t1, ft1, ft2            ; 0.9 <= 0.9 ... t1 should be 1

    li t2, 0x3dcccccd           ; 0.1
    fmvwx ft2, t2        
    fle t2, ft1, ft2            ; 0.9 > 0.1 ... t1 should be 0    

    slli t0, t0, 2              ; scale up
    slli t1, t1, 1              ; scale up
    add a0, t0, t1
    add a0, a0, t2              ; should be 6
