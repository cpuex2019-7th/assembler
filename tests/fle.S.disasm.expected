lui t1, 259686
addi t1, t1, 1638
fmvwx ft1, t1
lui t2, 263168
addi t2, t2, 0
fmvwx ft2, t2
fle t0, ft1, ft2
lui t2, 259686
addi t2, t2, 1638
fmvwx ft2, t2
fle t1, ft1, ft2
lui t2, 253133
addi t2, t2, -819
fmvwx ft2, t2
fle t2, ft1, ft2
slli t0, t0, 2
slli t1, t1, 1
add a0, t0, t1
add a0, a0, t2
